Sr2 merge input

anaisthrl opened the pull request.

LouisCorbet committed:
feat: allow txt device file export to xml

LouisCorbet committed:
feat: xml generation and save from txt file

LouisCorbet committed:
refactor: code clean

LouisCorbet committed:
Merge branch 'develop' of https://github.com/moukmessie/scada-v6 into sr2-xml

LouisCorbet committed:
refactor: code clean

LouisCorbet committed:
build: adjuted .dll copy target

LouisCorbet committed:
fix: display type in xml

LouisCorbet committed:
build: adapted assembly reference

LouisCorbet committed:
fix: adjusted element group criteria

LouisCorbet committed:
feat: added initial directory to save xml

moukmessie committed:
feat: Add or update CNLs after merging

moukmessie committed:
Fix: remove end cnls number input

LouisCorbet committed:
Merge branch 'sr2-xml-addcnl' of https://github.com/moukmessie/scada-v6 into sr2-xml

LouisCorbet committed:
feat: better xml group criterias

LouisCorbet committed:
feat: disable next btn while no file selected

moukmessie committed:
fix : update i18n file

moukmessie committed:
Fix: ResetCnlNums import cnl step 3

moukmessie committed:
feat: Import or update CNL, with optional file variable

moukmessie committed:
feat: add btnImport_Click event function

moukmessie committed:
add: radio button selection event

moukmessie committed:
feat: Add new merge CNL form

moukmessie committed:
feat: create cnl with file variable

moukmessie committed:
feat: add select format name cnl

moukmessie committed:
Fix: generate cnl from file

moukmessie committed:
refactor: Clean up and restructure code

anaisthrl committed:
fix: affichage variables dans merge

anaisthrl committed:
feat: pré sélection  "name format"

anaisthrl committed:
fix: display

anaisthrl committed:
fix: Orthographe

LouisCorbet commented on ScadaAdmin/OpenExtensions/ExtImport/Code/ExtensionPhrases.cs:
```diff
@@ -115,6 +120,14 @@ public static void Init()
 			ImportCnlsStep3 = dict["ImportCnlsStep3"];
 			ImportCnlsCompleted = dict["ImportCnlsCompleted"];
 
+			//TODO  : ADD Specific in dictionnary
```
Remove comment

LouisCorbet commented on ScadaAdmin/OpenExtensions/ExtImport/Controls/CtrlCnlImport3.cs:
```diff
@@ -1,244 +1,426 @@
+        {
+            List<Obj> listPrefix = prefixesAndSuffixes.Select(item => new Obj { ObjNum = item.ObjNum, Name = item.Name }).ToList();
+            cbBoxPrefix.Items.Clear();
+            //set display member and value member for combobox
```
remove

LouisCorbet commented on ScadaAdmin/OpenExtensions/ExtImport/Controls/CtrlCnlImport3.cs:
```diff
@@ -1,244 +1,426 @@
+            cbBoxSuffix.ValueMember = "ObjNum";
+            cbBoxSuffix.DisplayMember = "Name";
+
+            //pré selection de name format
```
remove

LouisCorbet commented on ScadaAdmin/OpenExtensions/ExtImport/Controls/CtrlCnlImport3.cs:
```diff
@@ -1,244 +1,426 @@
+            _mnemonique = columns[1];
+            _adress = columns[0];
+
+            string prefix = Regex.Split(_adress, @"[0-9]").First(); //Regex.Replace(_adress, @"[^0-9]", "");
```
remove comment

LouisCorbet commented on ScadaAdmin/OpenExtensions/ExtImport/Controls/CtrlCnlImport3.cs:
```diff
@@ -1,244 +1,426 @@
+
+            _adress = new string(_adress.SkipWhile(x => !char.IsDigit(x)).ToArray());
+
+            //setFormatType(columns[2]);
```
remove

LouisCorbet commented on ScadaAdmin/OpenExtensions/ExtImport/Controls/CtrlCnlImport3.cs:
```diff
@@ -1,244 +1,426 @@
+            //setFormatType(columns[2]);
+            _comment = columns[3].Replace("\"", "");
+
+            //add in dictionary
```
remove

LouisCorbet commented on ScadaAdmin/OpenExtensions/ExtImport/Controls/CtrlCnlImport3.cs:
```diff
@@ -1,244 +1,426 @@
+                _mnemonique = colums[0];
+                _adress = colums[1];
+                _adress = new string(_adress.SkipWhile(x => !char.IsDigit(x)).ToArray());
+                //setFormatType(colums[2]);
```
remove

LouisCorbet commented on ScadaAdmin/OpenExtensions/ExtImport/Controls/CtrlCnlImport3.cs:
```diff
@@ -1,244 +1,426 @@
+                _comment = colums[3].Replace("\"", "");
+                string prefix = Regex.Split(_adress, @"[0-9]").First();
+
+                //add in dictionary
```
remove

LouisCorbet commented on ScadaAdmin/OpenExtensions/ExtImport/Controls/CtrlCnlImport3.cs:
```diff
@@ -1,244 +1,426 @@
+
+                string prefix = Regex.Split(_adress, @"[0-9]").First();
+
+                //add in dictionary
```
remove

LouisCorbet commented on ScadaAdmin/OpenExtensions/ExtImport/Forms/FrmCnlImport.cs:
```diff
@@ -47,154 +58,160 @@ private void ApplyStep(int offset)
-					btnCreate.Enabled = true;//CtrlCnlImport3.FileSelected;
+					lblStep.Text = ExtensionPhrases.CreateCnlsStep3;
+					ctrlCnlmport3.Visible = true;
+					//chkPreview.Visible = true;
```
remove

LouisCorbet commented on ScadaAdmin/OpenExtensions/ExtImport/Forms/FrmCnlImport.cs:
```diff
@@ -47,154 +58,160 @@ private void ApplyStep(int offset)
+					btnCreate.Visible = true;
+					//if(ctrlCnlmport3.lastCheckState)
+					//	btnCreate.Visible = ctrlCnlmport3.FileSelected;//true;
+					//else
```
remove

LouisCorbet commented on ScadaAdmin/OpenExtensions/ExtImport/Forms/FrmCnlImport.cs:
```diff
@@ -47,154 +58,160 @@ private void ApplyStep(int offset)
-					AddChannels(cnls);
-					DialogResult = DialogResult.OK;
+
+					//remove cnls
```
remove

LouisCorbet commented on ScadaAdmin/OpenExtensions/ExtImport/Forms/FrmCnlsMerge.Designer.cs:
```diff
@@ -0,0 +1,287 @@
+{
+    partial class FrmCnlsMerge
+    {
+        //private DataGridView dataGridView;
```
remove

LouisCorbet commented on ScadaAdmin/OpenExtensions/ExtImport/Forms/FrmCnlsMerge.cs:
```diff
@@ -0,0 +1,593 @@
+    {
+
+        private Dictionary<string, List<string>> dictio;
+        private IAdminContext adminContext; // the Administrator context
+        private ScadaProject project;       // the project under development
```
remove comments

LouisCorbet commented on ScadaAdmin/OpenExtensions/ExtImport/Forms/FrmCnlsMerge.cs:
```diff
@@ -0,0 +1,593 @@
+                    {
+                        CnlNum = cnlNum,
+                        Active = cnlPrototype.Active,
+                        Name = namePrefix + cnlPrototype.Name,
+                        DataTypeID = cnlPrototype.DataTypeID,
+                        DataLen = cnlPrototype.DataLen,
+                        CnlTypeID = cnlPrototype.CnlTypeID,
+                        ObjNum = objNum,
+                        DeviceNum = deviceNum,
+                        TagNum = cnlPrototype.TagNum,
+                        TagCode = tagCode,
+                        FormulaEnabled = cnlPrototype.FormulaEnabled,
+                        InFormula = cnlPrototype.InFormula,
+                        OutFormula = cnlPrototype.OutFormula,
+                        FormatID = project.ConfigDatabase.GetFormatByCode(cnlPrototype.FormatCode)?.FormatID,
+                        QuantityID = project.ConfigDatabase.GetQuantityByCode(cnlPrototype.QuantityCode)?.QuantityID,
+                        UnitID = project.ConfigDatabase.GetUnitByCode(cnlPrototype.UnitCode)?.UnitID,
+                        LimID = null,
+                        ArchiveMask = cnlPrototype.ArchiveMask,
+                        EventMask = cnlPrototype.EventMask
+                    });
+
+                    cnlNum++; // Increment cnlNum for each element in store
+                }
+            }
+
+            return cnls;
+        }
+        public Dictionary<string, List<string>> StoreDictioData()
+        {
+            Dictionary<string, List<string>> resultDictionnary = new Dictionary<string, List<string>>();
+
+            // Parcourir dictio et ajouter des éléments à resultDictionnary
+            foreach (KeyValuePair<string, List<string>> kvp in dictio)
+            {
+                var list = new List<string>
+        {
+            kvp.Value[0], // Mnemonique
+            kvp.Value[1], // Type
+            kvp.Value[2]  // Descr
```
Prefer adding a single comment before the function than comment every line, except specific need

LouisCorbet commented on ScadaAdmin/OpenExtensions/ExtImport/Forms/FrmCnlsMerge.cs:
```diff
@@ -0,0 +1,593 @@
+                {
+                    currentCheckbox.Value = false;
+
+                    //color
```
remove

LouisCorbet commented on ScadaAdmin/OpenExtensions/ExtImport/Forms/FrmCnlsMerge.cs:
```diff
@@ -0,0 +1,593 @@
+                        otherCheckbox.Value = true;
+                        currentCheckbox.Value = false;
+
+                        //color
```
remove

LouisCorbet commented on ScadaAdmin/OpenExtensions/ExtImport/Forms/FrmCnlsMerge.cs:
```diff
@@ -0,0 +1,593 @@
+                        otherCheckbox.Value = false;
+                        currentCheckbox.Value = true;
+
+                        //color
```
remove

LouisCorbet commented on ScadaAdmin/OpenExtensions/ExtImport/Forms/FrmCnlsMerge.cs:
```diff
@@ -0,0 +1,593 @@
+                    {
+                        row.Cells[1].Value = true;
+
+                        //color
```
remove

LouisCorbet commented on ScadaAdmin/OpenExtensions/ExtImport/Forms/FrmCnlsMerge.cs:
```diff
@@ -0,0 +1,593 @@
+                    {
+                        row.Cells[1].Value = false;
+
+                        //color
```
remove

LouisCorbet commented on ScadaAdmin/OpenExtensions/ExtImport/Forms/FrmCnlsMerge.cs:
```diff
@@ -0,0 +1,593 @@
+                    {
+                        row.Cells[7].Value = true;
+
+                        //color
```
remove

LouisCorbet commented on ScadaAdmin/OpenExtensions/ExtImport/Forms/FrmCnlsMerge.cs:
```diff
@@ -0,0 +1,593 @@
+                    {
+                        row.Cells[7].Value = false;
+
+                        //color
```
remove

LouisCorbet commented on ScadaAdmin/OpenExtensions/ExtImport/Forms/FrmVariableMerge.Designer.cs:
```diff
@@ -0,0 +1,280 @@
+{
+    partial class FrmVariableMerge
+    {
+        //private DataGridView dataGridView;
```
remove

LouisCorbet commented on ScadaAdmin/OpenExtensions/ExtImport/Forms/FrmVariableMerge.cs:
```diff
@@ -0,0 +1,688 @@
+    {
+
+        private Dictionary<string, List<string>> dictio;
+        //private Dictionary<string, List<string>> _oldDictio = new Dictionary<string, List<string>>();
```
remove

LouisCorbet commented on ScadaAdmin/OpenExtensions/ExtImport/Forms/FrmVariableMerge.cs:
```diff
@@ -0,0 +1,688 @@
+        //private Dictionary<string, List<string>> _oldDictio = new Dictionary<string, List<string>>();
+        Dictionary<string, Dictionary<string, string>> _oldDictio = new Dictionary<string, Dictionary<string, string>>();
+
+        //private Dictionary<string, List<string>> _newDictio = new Dictionary<string, List<string>>();
```
remove

LouisCorbet commented on ScadaAdmin/OpenExtensions/ExtImport/Forms/FrmVariableMerge.cs:
```diff
@@ -0,0 +1,688 @@
+
+
+        private List<string> _listOfKey = new List<string>();
+        private IAdminContext adminContext; // the Administrator context
+        private ScadaProject project;       // the project under development
```
remove comments

LouisCorbet commented on ScadaAdmin/OpenExtensions/ExtImport/Forms/FrmVariableMerge.cs:
```diff
@@ -0,0 +1,688 @@
+            this.CtrlCnlImport1 = ctrlCnlImport1;
+            this.CtrlCnlImport2 = ctrlCnlImport2;
+            this.CtrlCnlImport3 = ctrlCnlImport3;
+            //this.CtrlCnlImport3 = ctrlCnlmport3;
```
remove

LouisCorbet commented on ScadaAdmin/OpenExtensions/ExtImport/Forms/FrmVariableMerge.cs:
```diff
@@ -0,0 +1,688 @@
+                    {
+                        row.Cells[1].Value = true;
+
+                        //color
```
remove

LouisCorbet commented on ScadaAdmin/OpenExtensions/ExtImport/Forms/FrmVariableMerge.cs:
```diff
@@ -0,0 +1,688 @@
+                    {
+                        row.Cells[1].Value = false;
+
+                        //color
```
remove

LouisCorbet commented on ScadaAdmin/OpenExtensions/ExtImport/Forms/FrmVariableMerge.cs:
```diff
@@ -0,0 +1,688 @@
+                    {
+                        row.Cells[6].Value = true;
+
+                        //color
```
remove

LouisCorbet commented on ScadaAdmin/OpenExtensions/ExtImport/Forms/FrmVariableMerge.cs:
```diff
@@ -0,0 +1,688 @@
+                    {
+                        row.Cells[6].Value = false;
+
+                        //color
```
remove

LouisCorbet commented on ScadaAdmin/OpenExtensions/ExtImport/Forms/FrmVariableMerge.cs:
```diff
@@ -0,0 +1,688 @@
+            };
+                    dico[key] = list;
+                }
+                else if (Convert.ToBoolean(row.Cells[6].Value) == true) // Si la case de droite est cochée
```
replace by a single comment before function

LouisCorbet commented on ScadaAdmin/OpenExtensions/ExtImport/Forms/FrmVariableMerge.cs:
```diff
@@ -0,0 +1,688 @@
+        {
+            Dictionary<string, List<string>> resultDictionnary = new Dictionary<string, List<string>>();
+
+            // Parcourir dictio et ajouter des éléments à resultDictionnary
```
replace by single comment before function or remove

LouisCorbet commented on ScadaAdmin/OpenExtensions/ExtImport/Forms/FrmVariableMerge.cs:
```diff
@@ -0,0 +1,688 @@
+            {
+                string address = Regex.Replace(kvp.Key, @"[^0-9]", "");
+                Dictionary<string, string> data = kvp.Value;
+                CnlPrototype cnlPrototype = ((List<CnlPrototype>)CtrlCnlImport1.CnlPrototypes)[0]; // Assuming there is at least one prototype
```
What if list is empty or null ?

LouisCorbet commented on ScadaAdmin/OpenExtensions/ExtImport/Forms/FrmVariableMerge.cs:
```diff
@@ -0,0 +1,688 @@
+                        break;
+                    }
+
+                    cnlNum += dataLength; // Increment cnlNum for each new address by the length of data
```
remove comment or add comment before function

LouisCorbet commented on ScadaAdmin/OpenExtensions/ExtImport/Forms/FrmVariableMerge.cs:
```diff
@@ -0,0 +1,688 @@
+
+
+                cnl.TagCode = address;
+                //cnl.DataTypeID = cnlDataType.TryGetValue(data["type"], out int dataTypeID);
```
remove

LouisCorbet commented on ScadaAdmin/OpenExtensions/ExtImport/Forms/FrmVariableMerge.cs:
```diff
@@ -0,0 +1,688 @@
+                {
+                    AddRowToDict(row, resultDict, 2, 3, 4, true);
+                }
+                else if (Convert.ToBoolean(row.Cells[6].Value)) // Si la case de droite est cochée
```
replace by single comment before function

LouisCorbet commented on ScadaAdmin/OpenExtensions/ExtImport/Forms/FrmVariableMerge.cs:
```diff
@@ -0,0 +1,688 @@
+                }
+            }
+
+            addChannelsAfterImport(); //création des channels après xml
```
replace comments by single comment before function

LouisCorbet reviewed (commented):
Why FrmCnlsMerge and FrmVariableMerge are almost the same ?

anaisthrl committed:
fix: remove comment

LouisCorbet commented on ScadaAdmin/OpenExtensions/ExtImport/Controls/CtrlCnlImport1.cs:
```diff
@@ -12,208 +15,207 @@
+
+        private void cbCommLine_SelectedIndexChanged(object sender, EventArgs e)
+        {
+            // filter devices by selected communication line
```
remove

LouisCorbet commented on ScadaAdmin/OpenExtensions/ExtImport/Controls/CtrlCnlImport1.cs:
```diff
@@ -12,208 +15,207 @@
+
+        private void cbDevice_SelectedIndexChanged(object sender, EventArgs e)
+        {
+            // create user interface of selected device
```
remove

LouisCorbet commented on ScadaAdmin/OpenExtensions/ExtImport/Forms/FrmCnlImport.cs:
```diff
@@ -47,154 +58,160 @@ private void ApplyStep(int offset)
+			ctrlCnlImport1.Visible = false;
+			ctrlCnlImport2.Visible = false;
+			ctrlCnlmport3.Visible = false;
+			//chkPreview.Visible = false;
```
remove

anaisthrl committed:
fix: remove comments

moukmessie committed:
feat(validation): refactor code and add checks with ScadaUiUtils warnings and errors

moukmessie committed:
feat: Update i18n and add warning boxes

- Update i18n implementation to allow for full localization of text elements including labels, butons and titles.
- Introduce warning message boxes for better user feedback and error handling.

This enhancement increases the accessibility of the application and user experience by supporting translation through the language XML file and providing clearer warnings to the user.
